components/playlist/browser/playlist_thumbnail_downloader.cc

```diff
@@ -59,84 +58,182 @@ void PlaylistThumbnailDownloader::DownloadThumbnail(
   CancelDownloadRequest(id);
 
   if (pause_download_for_testing_) {
-    ticket_map_[id] = {};
+    url_loader_map_[id] = {};
     return;
   }
 
-  auto ticket = request_helper_->Download(
-      thumbnail_url, {}, {}, target_thumbnail_path,
-      base::BindOnce(&PlaylistThumbnailDownloader::OnThumbnailDownloaded,
-                     base::Unretained(this), id),
-      {}, {.auto_retry_on_network_change = true});
-  ticket_map_[id] = ticket;
+  auto iter = CreateURLLoaderHandler(thumbnail_url);
+  iter->get()->DownloadToStringOfUnboundedSizeUntilCrashAndDie(
+      url_loader_factory_.get(),
+      base::BindOnce(&PlaylistThumbnailDownloader::SaveResponseToFile,
+                     weak_ptr_factory_.GetWeakPtr(), iter, id,
+                     target_thumbnail_path));
+  url_loader_map_[id] = iter;
+}
+
+void PlaylistThumbnailDownloader::DownloadThumbnail(
+    const std::string& id,
+    const GURL& thumbnail_url,
+    base::OnceCallback<void(gfx::Image)> callback) {
+  VLOG(2) << __func__ << " " << id << " : " << thumbnail_url.spec();
+  CancelDownloadRequest(id);
+
+  if (pause_download_for_testing_) {
+    url_loader_map_[id] = {};
+    return;
+  }
+
+  auto iter = CreateURLLoaderHandler(thumbnail_url);
+  iter->get()->DownloadToStringOfUnboundedSizeUntilCrashAndDie(
```

Comment:
We don't want to set a download size limit?

Reply:
For now, I'm not sure how big it should be ☺️

Reply:
Is there a disadvantage of setting something very high, like 5mb, just so a malicious endpoint can't crash the browser if they send 30gb of data?

Reply:
Yeah, makes sense. I googled what size is generally used for Youtube thumbnail and `og:tag image`. They say "1200~1280 * 720~1280". So 5mb would be good I think